### DISCUSSION TITLE
Woche 4

### DISCUSSION THREAD

**Discussion by lukashert** [Chrissi-Ruege/HealthKeeper#5] [General]
# Woche 4

Hallo zusammen!
In dieser Woche haben wir uns unsere Projektziele näher definiert, außerdem haben wir ein erstes Mockup für unser Frontend erstellt und uns mit den ersten Use Case Spezifikationen beschäftigt. 

## Projektziele

* **Grafiken**: Es werden Diagramme zum tracken von Gewicht, Kalorien und ähnlichem eingebunden.
* **Aktivitätentracker**: Auflistung der Sportaktivitäten mit angegebener Kalorienverbrennung
* **Ernährungstagebuch**: Tägliche Eingabe von Mahlzeiten mit ihren Kalorienwerten, Makros und Nährwerten
* **Wasserflasche**: Grafische Darstellung von getrunkenem Wasser mit Erinnerung 
* **Kalender**: Kalender mit Erinnerung an Sportterminen und Trainingsplänen
* **Trainingspläne und Video**s: Angebotene Fitness und Gesundheits Videos und Trainingspläne, angepasst auf Person und Suche
* **Konto-System**: Kontos erstellen und anmelden mit Fokus des Ziels (Abnehmen, Zunehmen, Gewicht halten mit Mengenangabe/Monat)

## Frontend
Wir haben in dieser Woche an einem erstes Mockup für unser Frontend gearbeitet. Dieses werden wir als Basis für die spätere Webseite verwendet.

<img width="775" alt="Screenshot 2023-11-06 152202" src="https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/02505e6e-08d7-462d-9163-da4bd538e8d3">

## Aktivitäts- und Sequenzdiagramme

Wir haben für drei Use Cases Aktivitäts- und Sequenzdiagramme gezeichnet.

### Create calendar entry
In diesen Diagrammen wird beschrieben, wie ein Benutzer einen neuen Kalender Eintrag anlegt.

![create-calendar-activity](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/212b3501-b942-4944-8f19-49af0fd86e06)
![create-calendar-sequence](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/58d4907d-451a-48f4-b7ba-c7440ddebae8)

### Login 
Die folgenden Diagramme beschreiben den Weg des Users von der Login Seite bis zur Startseite.

![login-activity](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/fed16089-6c56-452d-b41e-9589152813c5)
![login-sequence](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/9f284e7b-e1f1-454f-bda8-aca0ea42bed4)

### Reset password
Sollte ein Benutzer sein Passwort vergessen, sollte dieser es wieder zurücksetzen können.

![resetpassword-activity](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/6a09312e-2733-4158-880f-b7442952fc3e)
![resetpassword-sequence](https://github.com/Chrissi-Ruege/HealthKeeper/assets/20227840/147df3da-bb2e-4e01-9901-0918f29e9c0f)

## Weiterführende Links
* [SRS](https://github.com/Chrissi-Ruege/HealthKeeper/blob/main/Documentation/SRS.md)
* [UCRS Login](https://github.com/Chrissi-Ruege/HealthKeeper/blob/main/Documentation/UCRS-Login.md)
* [UCRS Reset password](https://github.com/Chrissi-Ruege/HealthKeeper/blob/main/Documentation/UCRS-Reset-password.md)

**Comment by NeeroxZ** [comment]
<img width="472" alt="image" src="https://github.com/Chrissi-Ruege/HealthKeeper/assets/84893432/6bd13344-289e-43b3-985d-c2edb607b239">
Leider kann man die Aktivitätsdiagramme nicht erkennen, falls es funktioniert: Siehe Screenshot

**Reply by Puggingtons** [reply]
wenn man nicht den dark mode benutzt, gehts 👌

**Reply by NeeroxZ** [reply]
Was diese "Kein Darkmode" ?

**Comment by Puggingtons** [comment]
Finds sehr gut, dass euer Projekt, auch neben den Aufgaben der Vorlesung, weiter entwickelt wird und fortschreitet 👍
In eurem Reset Password Sequenzdiagramm habt ihr um User Service If Abfragen. Im Diagramm sind aber jeweils nur die "ja" Pfade angegeben. Wie ist das Verhalten, wenn diese Abfragen nicht bestätigt werden?
